spec/text-document-spec.coffee

```diff
@@ -2,28 +2,51 @@ Point = require "../src/point"
 TextDocument = require "../src/text-document"
 
 describe "TextDocument", ->
-  document = null
-
-  beforeEach ->
-    document = new TextDocument
+  [document, layer] = []
 
   describe "::buildDisplayLayer()", ->
     describe "::slice(start, end)", ->
       it "returns the content between the start and end points", ->
+        document = new TextDocument(softWrapColumn: 10, tabLength: 4)
         document.setText("""
           abcd\tefg
 
           hij
         """)
-
-        layer = document.buildDisplayLayer(
-          softWrapColumn: 10
-          tabLength: 4
-        )
-
-        expect(layer.getLines()).toEqual [
+        expect(document.displayLayer.getLines()).toEqual [
           "abcd\t   "
           "efg\n"
           "\n"
           "hij"
         ]
+
+  describe "position translation", ->
+    beforeEach ->
+      document = new TextDocument(softWrapColumn: 100, tabLength: 2)
+      document.setText("zero\none\r\ntwo\nthree")
+
+    describe "::characterIndexForPosition(position)", ->
+      it "returns the absolute character offset for the given position", ->
+        expect(document.characterIndexForPosition(Point(0, 0))).toBe 0
```

Comment:
I understand that all the codebase basically uses arrays instead of `Point` objects, but could we somehow enforce using `Point` everywhere? In addition to being consistent, it also makes us save a few array allocations since we don't have to convert anything. Yes, it would be a breaking change but it should be super-easy to adopt this new constraint.